Automattic/wp-calypso
# Updates Upwork Banner to be independent of themes

#### Changes proposed in this Pull Request

This PR extracts the Upwork Banner functionality from the the Themes specific context. It also adds the an AB Test to show that banner on `/help`

![screen shot 2019-03-04 at 3 43 34 pm](https://user-images.githubusercontent.com/6817400/53764801-4847dc00-3e9c-11e9-8c53-fd89e2803334.png)

![image](https://user-images.githubusercontent.com/4924246/50303441-e7366d80-0441-11e9-8cf0-0fb88f118a4f.png)

![image](https://user-images.githubusercontent.com/4924246/50303414-dab21500-0441-11e9-83f0-a04a0e741761.png)

#### Themes Banner Testing instructions

1. Ensure that you are in the test. Set variation (builderReferralBanner): In console run `localStorage.setItem( 'ABTests', '{"builderReferralThemesBanner_20181218":"builderReferralBanner"}' );`
2. Go to `/themes`.
3. Check if you see the banner above the Search bar and make sure it links to the Upwork referral landing page.
4. Make sure that dismiss button closes banner

#### Testing Themes Banner Tracks Events

You can turn on analytics logging by running this in the console: localStorage.setItem( 'debug', 'calypso:analytics:*' )

- Refresh after running code to active analytics logging
- On viewing the banner: `calypso:analytics:tracks Record event Record event "calypso_upwork_banner_view" called with props {plan: "", location: "themes"}`
- On clicking CTA: `Record event "calypso_upwork_banner_start_now_button_click" called with props {plan: "", location: "themes"}`
- On dismissing: `Record event "calypso_upwork_banner_dismiss_icon_click" called with props {plan: "", location: "themes"}`

#### Help Banner Testing instructions

1. Ensure that you are in the test. Set variation (builderReferralBanner): In console run `localStorage.setItem( 'ABTests', '{"builderReferralHelpBanner_20190304":"builderReferralBanner"}' );`
2. Go to `/help`
3. Check if you see the banner above the Search bar and make sure it links to the Upwork referral landing page.
4. Make sure that dismiss button closes banner

#### Help Banner Tracks Events

- Refresh after running code to active analytics logging
- On viewing the banner: `calypso:analytics:tracks Record event Record event "calypso_upwork_banner_view" called with props {plan: "", location: "help-home"}`
- On clicking CTA: `Record event "calypso_upwork_banner_start_now_button_click" called with props {plan: "", location: "help-home"}`
- On dismissing: `Record event "calypso_upwork_banner_dismiss_icon_click" called with props {plan: "", location: "help-home"}`
